lib/galaxy/model/store/__init__.py

```diff
@@ -2215,6 +2217,12 @@ def add_dataset_collection(
     def add_dataset(self, dataset: model.DatasetInstance, include_files: bool = True) -> None:
         self.included_datasets[dataset] = (dataset, include_files)
 
+    def _ensure_dataset_file_accessible(self, dataset: model.DatasetInstance) -> None:
```

Comment:
```suggestion
    def _ensure_dataset_file_exists(self, dataset: model.DatasetInstance) -> None:
```

i guess accessible could be a permission level check